pkgs/by-name/ma/marble-shell-theme/package.nix

```
@@ -0,0 +1,47 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  python3,
+  colors ? [ "all" ], # Default to install all available colors
+  additionalInstallationTweaks ? [ ], # Additional installation tweaks
+}:
+assert colors != [ ];
+stdenv.mkDerivation {
+  pname = "marble-shell-theme";
+  version = "46.2.3";
+
+  src = fetchFromGitHub {
+    owner = "imarkoff";
+    repo = "Marble-shell-theme";
+    rev = "5971b15d8115c60c3a16b1d219ecffd2cfcdb323";
+    hash = "sha256-TX6BSS29EAi2PjL1fMvEKD12RjB9xrfqPSQsJJrUcJg=";
+  };
+
+  buildInputs = [ python3 ];
+
+  patchPhase = ''
+    runHook prePatch
+    substituteInPlace scripts/config.py \
+      --replace-fail "~/.themes" "/build/.themes"
+    runHook postPatch
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/themes
+    python install.py ${
+      lib.concatMapStringsSep " " (color: "--${color}") colors
+    } ${builtins.toString additionalInstallationTweaks}
+    cp -r /build/.themes/* $out/share/themes/
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Shell theme for GNOME DE";
+    license = lib.licenses.gpl3;
+    platforms = lib.platforms.linux;
+    homepage = "https://github.com/imarkoff/Marble-shell-theme";
```

Maybe wrap the derivation with `mkDerivation (finalAttrs: { ... })`.

```suggestion
    homepage = "https://github.com/imarkoff/Marble-shell-theme";
    changelog = "https://github.com/imarkoff/Marble-shell-theme/releases/tag/${finalAttrs.version}";
```